registry/handlers/httperror.go

```diff
@@ -0,0 +1,34 @@
+package handlers
+
+import (
+	"github.com/docker/distribution/registry/api/v2"
+	"net/http"
+)
+
+// HTTPError is a wrapper for v2.Errors that adds a Status int to hold
+// an HTTP Status code which will be used to set the status code on
+// the response.
+type HTTPError struct {
+	v2.Errors
```

Comment:
Looking at the code flow, I really don't get why `HTTPError` has to have a `v2.Errors`. Its tightly coupled everything to `HTTPError`. We really need to be leverage type switching and interfaces or this is going to get out of hand.